src/Build/BuildCheck/OM/ProjectImportedCheckData.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Build.Experimental.BuildCheck;
+
+namespace Microsoft.Build.Experimental.BuildCheck;
+
+public class ProjectImportedCheckData : CheckData
```

Comment:
This will need xml doc comment (it's intended to be a public API)

Same for the public members.